src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,144 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static StorageDao storageDao;
+    private static RegistrationService registrationService;
+    private User mark;
+
+    @BeforeAll
+    static void beforeAll() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        mark = new User("Mark","123456", 18);
+    }
+
+    @Test
+    void register_setNullUser_NotOk() {
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void register_setNullLogin_NotOk() {
+        mark.setLogin(null);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(mark);
+        });
+    }
+
+    @Test
+    void register_setNullAge_NotOk() {
+        mark.setAge(null);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(mark);
+        });
+    }
+
+    @Test
+    void register_setNullPassword_NotOk() {
+        mark.setPassword(null);
+        assertThrows(RuntimeException.class, () -> {
```

Comment:
You don’t have this conditional, so now you  testing system exception

Reply:
Ooo, my fault, I actually used to do it according to the principle of TDD and just forgot